wayland-sys/src/client.rs

```diff
@@ -81,17 +81,30 @@ external_library!(WaylandClient, "wayland-client",
 #[cfg(feature = "dlopen")]
 lazy_static!(
     pub static ref WAYLAND_CLIENT_OPTION: Option<WaylandClient> = {
-        match WaylandClient::open("libwayland-client.so") {
-            Ok(h) => Some(h),
-            Err(::dlib::DlError::NotFound) => None,
-            Err(::dlib::DlError::MissingSymbol(s)) => {
-                if ::std::env::var_os("WAYLAND_RS_DEBUG").is_some() {
-                    // only print debug messages if WAYLAND_RS_DEBUG is set
-                    eprintln!("[wayland-client] Found library libwayland-client.so cannot be used: symbol {} is missing.", s);
+        // This is a workaround for Ubuntu 17.04, which doesn't have a bare symlink
+        // for libwayland-client.so but does have it with the version numbers for
+        // whatever reason.
+        //
+        // We could do some trickery with str slices but that is more trouble
+        // than its worth
+        let versions = ["libwayland-client.so",
+                        "libwayland-client.so.0",
+                        "libwayland-client.so.0.3.0"];
+        let mut ret = None;
```

Comment:
I don't think this dans with `ret` is necessary. We can just early-return at the first `Ok(_)`.

Reply:
Huh, you're right. I figured that I wouldn't be able to do it since that code block isn't really a function but I guess I was wrong